lib/cli-update-connection.ts

```diff
@@ -27,58 +27,66 @@ export default function CliUpdateConnection() {
     program.parse(process.argv);
     let settings = config.merge(program);
 
-    if (process.argv.length === 2) {
+    if (process.argv.length === 2 || !settings.dataset) {
         program.help();
     } else {
         try {
             let credentials = new msrest.TokenCredentials(settings.accessKey, 'AppKey');
             let client = new powerbi.PowerBIClient(credentials, settings.baseUri, null);
 
-            client.datasets.getDatasetById(settings.collection, settings.workspace, settings.dataset, (err, result) => {
-                if (err) {
-                    return cli.error(err);
+            client.datasets.getDatasetById(settings.collection, settings.workspace, settings.dataset, (getDatasetError, result) => {
+                if (getDatasetError) {
+                    return cli.error(getDatasetError);
                 }
 
                 cli.success('Found dataset!');
                 cli.print('Id: %s', result.id);
                 cli.print('Name: %s', result.name);
 
                 if (settings.connectionString) {
-                    updateConnectionString(client, settings);
+                    updateConnectionString(client, settings, function (updateConnectionStringError) {
+                        if (updateConnectionStringError) {
+                            return cli.error(updateConnectionStringError);
+                        }
+                    });
                 }
 
                 if (settings.username && settings.password) {
-                    updateCredentials(client, settings)
+                    updateCredentials(client, settings, function (updateCredentialsError) {
+                        if (updateCredentialsError) {
+                            return cli.error(updateCredentialsError);
+                        }
+                    });
                 }
             });
         } catch (err) {
             cli.error(err);
         }
     }
 
-    function updateConnectionString(client: powerbi.PowerBIClient, settings: any, callback?): void {
+    function updateConnectionString(client: powerbi.PowerBIClient, settings: any, callback: (err: Error, result?: any) => void): void {
         let params: { [propertyName: string]: any } = {
             connectionString: settings.connectionString
         };
 
+        cli.print('Updating connection string...');
         client.datasets.setAllConnections(settings.collection, settings.workspace, settings.dataset, params, (err, result) => {
             if (err) {
-                cli.error('Error updating connection string');
-                cli.error(err);
-                return;
+                callback(err);
             }
 
             cli.success('Connection string successfully updated');
             cli.print('Dataset: ', settings.dataset);
             cli.print('ConnectionString: ', settings.connectionString);
 
-            if (callback) {
+            if (result) {
```

Comment:
It seems we should invoke the callback to signal completion even if the result was null. Either callback(null, result) without the conditional, and let the caller can figure out what to do with a null result, or if a null result should be treated as an error then invoke with a new error that indicates this state?